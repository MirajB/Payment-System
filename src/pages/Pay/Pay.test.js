import PayPage from './index';
import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => jest.fn(),
}));

it('renders payment page', () => {
  const { container } = render(<PayPage />);
  expect(container).toMatchSnapshot();
});

it('renders credit card screen when clicked', () => {
  const route = '/pay';
  const { container } = render(
    <MemoryRouter initialEntries={[route]}>
      <PayPage />
    </MemoryRouter>
  );
  const ccButton = screen.getByText('Credit Card');
  fireEvent.click(ccButton);
  expect(container).toMatchSnapshot();
});

it('renders debit card screen when clicked', () => {
  const route = '/pay';
  const { container } = render(
    <MemoryRouter initialEntries={[route]}>
      <PayPage />
    </MemoryRouter>
  );
  const ccButton = screen.getByText('Debit Card');
  fireEvent.click(ccButton);
  expect(container).toMatchSnapshot();
});
it('renders net banking screen when clicked', () => {
  const route = '/pay';
  const { container } = render(
    <MemoryRouter initialEntries={[route]}>
      <PayPage />
    </MemoryRouter>
  );
  const ccButton = screen.getByText('Net Banking');
  fireEvent.click(ccButton);
  expect(container).toMatchSnapshot();
});

it('Can cancel the payment', async () => {
  const route = '/pay';
  const { container } = render(
    <MemoryRouter initialEntries={[route]}>
      <PayPage />
    </MemoryRouter>
  );
  const ccButton = screen.getByText('Credit Card');
  fireEvent.click(ccButton);
  const cancelButton = screen.getByText('Cancel');
  fireEvent.click(cancelButton);
  expect(container).toMatchSnapshot();
});
