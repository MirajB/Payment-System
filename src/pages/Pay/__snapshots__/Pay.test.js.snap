// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can cancel the payment 1`] = `
<div>
  <h2>
    Select Mode of Payment
  </h2>
  <div>
    <button>
      Credit Card
    </button>
    <button>
      Debit Card
    </button>
    <button>
      Net Banking
    </button>
  </div>
</div>
`;

exports[`renders credit card screen when clicked 1`] = `
<div>
  <h2>
    Select Mode of Payment
  </h2>
  <div>
    <form
      action="#"
    >
      <label
        for="name"
      >
        Name
      </label>
      <input
        class="text-input"
        name="name"
        readonly=""
        type="text"
        value="undefined undefined"
      />
      <label
        for="your_emi"
      >
        Amount to be paid
      </label>
      <input
        class="text-input"
        name="your_emi"
        readonly=""
        type="text"
        value="undefined"
      />
      <label
        for="bankName"
      >
        Bank Name
      </label>
      <input
        class="text-input"
        name="bankName"
        placeholder="Enter bank name"
        type="text"
        value=""
      />
      <div
        class="css-1iwcm1"
      >
        <button
          type="submit"
        >
          Pay
        </button>
        <button>
          Cancel
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`renders debit card screen when clicked 1`] = `
<div>
  <h2>
    Select Mode of Payment
  </h2>
  <div>
    <form
      action="#"
    >
      <label
        for="name"
      >
        Name
      </label>
      <input
        class="text-input"
        name="name"
        readonly=""
        type="text"
        value="undefined undefined"
      />
      <label
        for="your_emi"
      >
        Amount to be paid
      </label>
      <input
        class="text-input"
        name="your_emi"
        readonly=""
        type="text"
        value="undefined"
      />
      <label
        for="bankName"
      >
        Bank Name
      </label>
      <input
        class="text-input"
        name="bankName"
        placeholder="Enter bank name"
        type="text"
        value=""
      />
      <div
        class="css-1iwcm1"
      >
        <button
          type="submit"
        >
          Pay
        </button>
        <button>
          Cancel
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`renders net banking screen when clicked 1`] = `
<div>
  <h2>
    Select Mode of Payment
  </h2>
  <div>
    <form
      action="#"
    >
      <label
        for="name"
      >
        Name
      </label>
      <input
        class="text-input"
        name="name"
        readonly=""
        type="text"
        value="undefined undefined"
      />
      <label
        for="your_emi"
      >
        Amount to be paid
      </label>
      <input
        class="text-input"
        name="your_emi"
        readonly=""
        type="text"
        value="undefined"
      />
      <label
        for="bankName"
      >
        Bank Name
      </label>
      <input
        class="text-input"
        name="bankName"
        placeholder="Enter bank name"
        type="text"
        value=""
      />
      <div
        class="css-1iwcm1"
      >
        <button
          type="submit"
        >
          Pay
        </button>
        <button>
          Cancel
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`renders payment page 1`] = `
<div>
  <h2>
    Select Mode of Payment
  </h2>
  <div>
    <button>
      Credit Card
    </button>
    <button>
      Debit Card
    </button>
    <button>
      Net Banking
    </button>
  </div>
</div>
`;
